### PR TITLE
Replace RepoFileNotFound with NotFoundException

### PR DESCRIPTION
**Pull request by phenaproxima**
We have two exceptions, `RepoFileNotFound` and `NotFoundException`, which serve very similar purposes. We should consolidate them. This converts all usages of `RepoFileNotFound` to `NotFoundException`, and removes `RepoFileNotFound`.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
         run: composer test
 
       - name: Check dependencies for known security vulnerabilities
-        run: composer audit
+        run: composer audit --no-dev
```
Comment (tedbow) [inline .github/workflows/build.yml:71]: How is this related?

**Review by tedbow — APPROVED** [review]
Looks good

File: .github/workflows/build.yml (hunk 1 above)
Comment (phenaproxima) [reply .github/workflows/build.yml:71]: Proof of failure: https://github.com/php-tuf/php-tuf/actions/runs/8266701537/job/22615313427#step:13:1
